impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js

```diff
@@ -45,14 +47,27 @@ define([
       return fontFamily;
     },
 
+    _getModelSize: function() {
+      var model = this.model;
+      return {
+        height: model.height || 400,
+        width: model.width || 400
+      };
+    },
+
     _initializeChart: function() {
-      this._chart = echarts.init(this.domContainer, null, {});
+      var size = this._getModelSize();
+      this._chart = echarts.init(this.domContainer, null, {width: size.width, height: size.height});
       return this._chart;
     },
 
     /** @override */
     _updateSize: function() {
-      this._chart.resize();
+      var size = this._getModelSize();
+      var clientHeight = this._chart.getDom().clientHeight;
+      var height = Math.max(clientHeight, size.height);
```

Comment:
This doesn't look right. And there's no comment explaining the reason for this logic.
If there is a bug affecting the values of model width and model height, we should fix that instead.
If this is a workaround until that bug is fixed in another JIRA issue, then we could link to that issue in a comment explaining the workaround. The JIRA issue would also point to the workaround piece of code so that when fixed, the workaround would also be removed.

Reply:
Will look into it and do the needful.